acceptancetests/assess_deploy_webscale.py

```diff
@@ -225,12 +235,43 @@ def parse_args(argv):
     return parser.parse_args(argv)
 
 
+def check_for_mongo_snap(args):
+    setattr(args, "db_snap_asserts_path", "/tmp/empty.asserts")
+    if args.db_snap_path:
+        # NOTE(achilleasa): ideally we should be using client.enable_feature()
+        # but it looks like the current implementation does not pass the
+        # enabled features to the backend; the backend however supports
+        # fetching the features through the JUJU_DEV_FEATURE_FLAGS envvar.
+        log.info("Enabling 'mongodb-snap' feature flag")
+        os.environ["JUJU_DEV_FEATURE_FLAGS"] = "mongodb-snap"
+
+        # Fetch snap if using a remote URL. Juju expects the snap file to match
+        # "juju-db_\d+.snap" so we should rename it accordingly.
+        if urlparse(args.db_snap_path).scheme != "":
+            snap_dst = "/tmp/juju-db_1.snap"
+            log.info("Downloading snap from {} as {}".format(
+                args.db_snap_path, snap_dst))
+            urlretrieve(args.db_snap_path, snap_dst)
+            setattr(args, "db_snap_path", snap_dst)
+
+        # Create empty assert file and enable appropriate feature flags
+        os.close(os.open(args.db_snap_asserts_path, os.O_CREAT | os.O_APPEND))
+
+    return args
+
+
 def main(argv=None):
     args = parse_args(argv)
     configure_logging(args.verbose)
+    args = check_for_mongo_snap(args)
+
     begin = time.time()
     bs_manager = BootstrapManager.from_args(args)
-    with bs_manager.booted_context(args.upload_tools):
+    with bs_manager.booted_context(
+        args.upload_tools,
+        db_snap_path=args.db_snap_path,
+        db_snap_asserts_path=args.db_snap_asserts_path
```

Comment:
So in general I'm usually a -1 on modifying args. I'd prefer if the function read args and then returned the result noting the db_snap name and the asserts path required. The issue is that it raises potential conflict for args in the future and makes it unclear if something is on the args object came from the user or the various code that modifies it. I tend to think of args like a user supplied chunk of data that is used but not modified so there's no doubt about the sources of things.

Reply:
Now that I think about it there is really no need to modify `args`. In some of my earlier attempts I was passing the args to the bootstrap manager c-tor and I forgot to clean up my code. I will push an update for this.